server/docs/getting-started.md

```diff
@@ -142,384 +288,97 @@ async function initWeb3() {
-- The contract generates die rolls and distributes money to the winner
-
-The information above is enough to determine the ActionType, Stage, AppState, and Action for the game.
-
-
-```solidity
-contract HighRollerApp is CounterfactualApp {
-
-  enum ActionType {
-    START_GAME,
-    COMMIT_TO_HASH,
-    COMMIT_TO_NUM,
-    REVEAL
-  }
-
-  enum Stage {
-    PRE_GAME,
-    COMMITTING_HASH,
-    COMMITTING_NUM,
-    REVEALING,
-    DONE
-  }
-
-  enum Player {
-    FIRST,
-    SECOND
-  }
-
-  struct AppState {
-    address[2] playerAddrs;
-    Stage stage;
-    bytes32 salt;
-    bytes32 commitHash;
-    uint256 playerFirstNumber;
-    uint256 playerSecondNumber;
-  }
-
-  struct Action {
-    ActionType actionType;
-    uint256 number;
-    bytes32 actionHash;
-  }
-```
-
-Because it’s such a simple game (player1 → player2 → player1 → done), getTurnTaker() and isStateTerminal() are both quite simple.
-
-```solidity
-function isStateTerminal(AppState memory state)
-  public
-  pure
-  returns (bool)
-{
-  return state.stage == Stage.DONE;
-}
-
-function getTurnTaker(AppState memory state)
-  public
-  pure
-  returns (Player)
-{
-  return state.stage == Stage.COMMITTING_NUM ? Player.SECOND : Player.FIRST;
+    web3Provider = window.web3.currentProvider;
+  }
+  // If no injected web3 instance is detected, fall back to Ganache
+  else {
+    web3Provider = new Web3.providers.HttpProvider('http://localhost:8545');
+  }
+  web3 = new Web3(web3Provider);
 }
 ```
 
-Actions for the game are defined in the applyAction function
+----------
 
-```solidity
-function applyAction(AppState memory state, Action memory action)
-  public
-  pure
-  returns (bytes memory)
-{
-  AppState memory nextState = state;
-  if (action.actionType == ActionType.START_GAME) {
-    require(
-      state.stage == Stage.PRE_GAME,
-      "Cannot apply START_GAME on PRE_GAME"
-    );
-    nextState.stage = Stage.COMMITTING_HASH;
-  } else if (action.actionType == ActionType.COMMIT_TO_HASH) {
-    require(
-      state.stage == Stage.COMMITTING_HASH,
-      "Cannot apply COMMIT_TO_HASH on COMMITTING_HASH"
-    );
-    nextState.stage = Stage.COMMITTING_NUM;
+## initContract()
 
-    nextState.commitHash = action.actionHash;
-  } else if (action.actionType == ActionType.COMMIT_TO_NUM) {
-    require(
-      state.stage == Stage.COMMITTING_NUM,
-      "Cannot apply COMMITTING_NUM on COMMITTING_NUM"
-    );
-    nextState.stage = Stage.REVEALING;
+This is where you'll create a TruffleContract object corresponding to the **HighRoller.sol** solidity contract. When developing your own apps, you'll use Truffle to point to your contract as you develop it. Let's fill some of the details in now.
 
-    nextState.playerSecondNumber = action.number;
-  } else if (action.actionType == ActionType.REVEAL) {
-    require(
-    state.stage == Stage.REVEALING,
-    "Cannot apply REVEALING on REVEALING"
-    );
-    nextState.stage = Stage.DONE;
+```typescript
```

Comment:
A high level comment I have here is that maybe we should start with the "main" function that uses all of these helper functions first and explain the main flow of `app.js` _first_ and then dive into the details of the helper functions?